### PR TITLE
Issue/31/add checkout entry to git externals info

### PR DESCRIPTION
**Pull request by arl**
* Add `Checkout` field to `git externals info` output. It represents the absolute path of the target repository
* Refactoring
* Remove `git externals list` command
* Enable the possibility to show the info of only one (or more) top-level external. The matching is performed on the `name` field if present or the last component of the externals path (ex: `.git/externals/foo/bar` would be `bar`)

### CODE REVIEW

**Comment by arl** [comment]
Fixes #31

**Comment by arl** [comment]
@naufraghi As soon as this is OK and merged, I'll push 2 commits on `deploy_qt5` that are required for the *plotter*:
 - use of the new Checkout field
 - fix with git submodules

File: git_externals/git_externals.py [hunk 1]
```diff
@@ -396,39 +396,43 @@ def autosvn_update_checkout(reset):
                 gitext_up(recursive, entries, reset=reset, use_gitsvn=use_gitsvn)
 
 
-def gitext_recursive_info(root_dir):
-    git_exts = load_gitexts()
-    git_exts = {ext_repo: ext for ext_repo, ext in git_exts.items()
+def gitext_recursive_info(root_dir, recursive=True, externals=[]):
+    git_exts = {ext_repo: ext for ext_repo, ext in load_gitexts().items()
```
Comment (naufraghi) [inline git_externals/git_externals.py:400]: What are we skipping here? Do you think it's better logging to the user or documenting in code what is happening?

File: git_externals/git_externals.py [hunk 2]
```diff
@@ -396,39 +396,43 @@ def autosvn_update_checkout(reset):
...
+            if recursive:
+                for dsts in ext['targets'].values():
+                    for dst in dsts:
+                        real_dst = os.path.realpath(os.path.join(cwd, dst))
 
-                    has_deps = any([os.path.realpath(d).startswith(real_dst)
-                                    for e in filtered.values()
-                                    for ds in e['targets'].values()
-                                    for d in ds])
+                        has_deps = any([os.path.realpath(d).startswith(real_dst)
```
Comment (naufraghi) [inline git_externals/git_externals.py:423]: What are we listing here in this nested for? Can you document a bit?

File: git_externals/git_externals.py [hunk 3]
```diff
@@ -396,39 +396,43 @@ def autosvn_update_checkout(reset):
...
+                                        for d in ds])
 
-                    if has_deps:
-                        gitext_recursive_info(os.path.join(root_dir, dst))
+                        if has_deps:
+                            gitext_recursive_info(os.path.join(root_dir, dst))
 
 
-def print_gitext_info(ext_repo, ext, root_dir):
+def print_gitext_info(ext_repo, ext, root_dir, checkout=None):
```
Comment (naufraghi) [inline git_externals/git_externals.py]: Better boolean? `checkout=False`

**Review by naufraghi — APPROVED** [review]
Did you test the code with a recursive repo?

File: git_externals/git_externals.py (hunk 1 above)
Comment (arl) [reply git_externals/git_externals.py:400]: This is a cosmetic change :
```python
git_exts = load_gitexts()
git_exts = {ext_repo: ext for ext_repo, ext in git_exts.items()		
                  if os.path.exists(os.path.join(externals_root_path(), get_repo_name(ext_repo)))}
```
into:
```python
git_exts = {ext_repo: ext for ext_repo, ext in load_gitexts().items()		
                  if os.path.exists(os.path.join(externals_root_path(), get_repo_name(ext_repo)))}
```

File: git_externals/git_externals.py (hunk 2 above)
Comment (arl) [reply git_externals/git_externals.py:423]: I didn't touch this code, just added the comment saying that here we start recursing into nested externals

**Comment by arl** [comment]
`git externals info` has been tested succesfully on a nested repository by @faism . The mirror was not working though, but it was probably because of the regression on the `name` field support

**Pull request merged**